plugins/moduleLoader.js

```diff
@@ -2,89 +2,125 @@
  * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
  * See LICENSE for license details.
  */
+/* eslint-disable global-require */
+/* eslint-disable import/no-dynamic-require */
 import Vue from 'vue'; // eslint-disable-line import/no-extraneous-dependencies
-// import { modules } from '@Root/modules.config';
+import additionalConfig from '@Root/modules.config'; // User modules configuration file
 
-const modules = [];
+class ModuleLoader {
+    constructor() {
+        this.config = this.setModulesConfiguration();
+        this.pagesConfig = this.setPagesConfiguration(this.config);
+        this.componentsConfig = this.setComponentsConfiguration(this.config);
+    }
+
+    get getPagesConfig() {
+        return this.pagesConfig;
+    }
+
+    get getComponentsConfig() {
+        return this.componentsConfig;
+    }
 
-const ModuleLoader = {
     install() {
```

Comment:
It will be clearer if you'll use Vue instance reference from arguments in vue plugin install method instead of modifying imported object:
install(Vue, [options]) {...}